src/geop/main.py

```diff
@@ -1,11 +1,60 @@
-from flask import Flask
+import os
+import time
+
+from flask import Flask, request, jsonify
+import numpy as np
+import rasterio
 
 app = Flask(__name__)
 
+DATA_PATH = '/usr/data/'
+
+
+@app.route('/counts', methods=['POST'])
+def count():
+    """
+    Perform a rudimentary cell count analysis on a portion of a
+    provided raster
+
+    Query String Args:
+        filename: file name of a valid geotiff in DATA_DIR
+    """
+
+    raster = request.args.get('filename')
+    if not raster:
+        return handle_error('filename parameter is required')
+
+    raster_path = os.path.join(DATA_PATH, raster)
+    if not os.path.isfile(raster_path):
+        return handle_error('{} is not valid file in DATA_DIR'.format(raster))
+
+    counts = {}
+    start = time.clock()
+    with rasterio.open(raster_path) as src:
+        w = src.read(1, window=((10000, 10350), (10000, 10350)))
+
+        for idx, cell in np.ndenumerate(w):
```

Comment:
FYI, doing some research into numpy is showing that this can be sped up by orders of magnitude, consider this just a first pass.